### PR TITLE
fix: return empty results on empty array filters

### LINKED ISSUE #757: Params with `_in` like `proposal_in` should not be empty arrays
Right now we return a response even if the user passes empty arrays, I think we should return an error in such cases  

Example query 
```Graphql
query Votes($voter: String!, $proposals: [String]!) {
    votes(first: 1000, where: { voter: $voter, proposal_in: $proposals }) {
      proposal {
        id
        space {
          id
        }
      }
    }
  }
 
  ```
variables
```JSON
{
  "voter": "0x24F15402C6Bb870554489b2fd2049A85d75B982f",
  "proposals": []
}
```

### PR DESCRIPTION
**Pull request by wa0x6e**
Fixes #757 

Return an empty response when array filter is empty
Retry of https://github.com/snapshot-labs/snapshot-hub/pull/763 which was reverted

Test with 

```graphql
query Spaces {
  spaces(first: 5, skip: 0, where: {id_in: []}) {
    name
    created
  }
}

```

Should return 

```json
{
  "errors": [
    {
      "message": "id_in must have at least one item",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "spaces"
      ]
    }
  ],
  "data": {
    "spaces": null
  }
}
```

Additionally, move all `throw new Error` to `throw new PulbicError` for consistency

# TODO

- [ ] Merge https://github.com/snapshot-labs/snapshot/pull/4418 first

### CODE REVIEW

File: src/graphql/helpers.ts [hunk 1]
```diff
@@ -114,16 +115,24 @@ export function buildWhereQuery(fields, alias, where) {
       params.push(fieldNot);
     }
 
-    const fieldIn = where[`${field}_in`] || [];
-    if (fieldIn.length > 0) {
-      query += `AND ${alias}.${field} IN (?) `;
-      params.push(fieldIn);
+    const fieldIn = where[`${field}_in`];
+    if (Array.isArray(fieldIn)) {
```
Comment (ChaituVR) [inline src/graphql/helpers.ts:119]: Looks like `_in`s are always array, we don't need to check

even if we query
```
query Votes {
    spaces(first: 1000, where: { id_in: "fabien.eth" }) {
      id
    }
  }
  ```
  
  `id_in` will be an array

Comment (ChaituVR) [reply src/graphql/helpers.ts:119]: I mean can be like before: 
```
    const fieldIn = where[`${field}_in`] || [];
```

Comment (wa0x6e) [reply src/graphql/helpers.ts:119]: > I mean can be like before:
> 
> ```
>     const fieldIn = where[`${field}_in`] || [];
> ```

`where[`${field}_in`]` can be either 
- something valid (continue), 
- an empty array (stop), 
- or undefined (ignore)

If we cast with `|| []`, example on the space query with `id` and `created` filter:

```
spaces(first: 1000, where: { id_in: "fabien.eth" })
```

will be considered as 

```
spaces(first: 1000, where: { id_in: "fabien.eth", created_in: [] })
``` 

and will always return nothing

Comment (ChaituVR) [reply src/graphql/helpers.ts:119]: Missed this point, thank you!

**Pull request merged**